packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2565,31 +2565,35 @@ class RenderPointerListener extends RenderProxyBoxWithHitTestBehavior {
   MouseTrackerAnnotation get hoverAnnotation => _hoverAnnotation;
 
   void _updateAnnotations() {
-    assert(_onPointerEnter != _hoverAnnotation.onEnter || _onPointerHover != _hoverAnnotation.onHover || _onPointerExit != _hoverAnnotation.onExit,
```

Comment:
Surprised that it's OK to remove this kind of assertion; I guess this method's role has changed...